aydin/nn/models/utils/masking/random_masked_dataset.py

```diff
@@ -0,0 +1,61 @@
+import numpy
+import torch
+from torch.utils.data import Dataset
+
+
+class RandomMaskedDataset(Dataset):
+    def __init__(
+        self,
+        image,
+        nb_masks=4,
+        pixel_masking_probability: float = 0.3,
+    ):
+        """
+
+        Parameters
+        ----------
+        image
+        """
+
+        self.image = torch.tensor(image)
+        self.nb_masks = nb_masks
+        self.p = pixel_masking_probability
+
+    def __len__(self):
+        return self.nb_masks
+
+    def get_mask(self, i):
```

Comment:
Do you need to pass `i` in this method?

Reply:
at the moment we don't need it honestly, I was originally thinking if we want to shift a random mask with different phases but maybe a bad idea, if we need it later we can add it later, i will remove it for now.